cldfbench_segbo.py

```diff
@@ -2,44 +2,112 @@
+        # values.csv
+        ds.remove_columns('ValueTable', 'Code_ID', 'Comment')
+        ds.add_columns(
+            'ValueTable',
+            {
+                "dc:extent": "singlevalued",
+                "datatype": "string",
+                # FIXME: ValueError: Duplicate column property: http://cldf.clld.org/v1.0/terms.rdf#languageReference
+                # "propertyUrl": "http://cldf.clld.org/v1.0/terms.rdf#languageReference",
+                "required": True,
+                "name": "Source_Language_ID",
+            },
+            'Inventory_ID',
+            *self.valueTableProperties,
+        )
+
+        # parameters.csv
+        table = ds.add_component('ParameterTable')
+
+        # languages.csv
+        ds.add_component('LanguageTable', *self.languageTableProperties)
+
+        # inventories.csv
         table = ds.add_table(
-            'segbo_with_glottolog.csv',
+            'inventories.csv',
             {'name': 'ID', 'propertyUrl': "http://cldf.clld.org/v1.0/terms.rdf#id"},
-            {'name': 'InventoryID'},
+            {'name': 'Language_ID', 'propertyUrl': "http://cldf.clld.org/v1.0/terms.rdf#languageReference"},
+            {'name': 'Language_Name', 'propertyUrl': "http://cldf.clld.org/v1.0/terms.rdf#name"},
+            {'name': 'Source', 'propertyUrl': 'http://cldf.clld.org/v1.0/terms.rdf#source', 'separator': ';'},
         )
         table.tableSchema.primaryKey = ['ID']
+        ds.add_foreign_key('ValueTable', 'Inventory_ID', 'inventories.csv', 'ID')
 
     def cmd_makecldf(self, args):
-        """
-        Convert the raw data to a CLDF dataset.
-
-        >>> args.writer.objects['LanguageTable'].append(...)
-        """
         self.create_schema(args.writer.cldf)
 
+        # values.csv
         counter = 1
         for row in self.raw_dir.read_csv(
-            self.raw_dir / 'segbo' / 'data' / 'segbo_with_glottolog.csv',
+            self.raw_dir / 'segbo' / 'data' / 'SegBo database - Phonemes.csv',
             dicts=True,
         ):
-            value = {
-                **{"ID": str(counter)},
-                **dict(row)
-            }
-            args.writer.objects['segbo_with_glottolog.csv'].append(value)
+            args.writer.objects['ValueTable'].append({
+                'ID': str(counter),
+                'Parameter_ID': str(counter),
+                'Inventory_ID': row['InventoryID'],
+                'Language_ID': row['BorrowingLanguageGlottocode'],
+                'Source_Language_ID': row['SourceLanguageGlottocode'],
+                'Inventory_ID': row['InventoryID'],
+                'Value': row['BorrowedSound'],
+                'Source': '', # TODO 
+                **{ k: row[k] for k in self.valueTableProperties}
+            })
+            # parameters.csv
+            args.writer.objects['ParameterTable'].append({
+                'ID': str(counter),
+                'Name': row['BorrowedSound'],
+                'Description': '', # TODO
```

Comment:
Ideally, this would have a column with `propertyUrl` https://cldf.clld.org/v1.0/terms.rdf#cltsReference - linking sounds to CLTS as far as possible.

Reply:
How could this be done? Otherwise I will just follow the way you did it in phoible